src/main/java/org/candlepin/controller/ConsumerManager.java

```diff
@@ -69,4 +92,99 @@ public void updateLastCheckIn(Consumer consumer) {
+     *  environment IDs is unknown to the owner, or if the target environments contains null or a duplicate
+     *  value.
+     *
+     * @return
+     *  the UUIDs of the {@link Consumer}s that were updated and had their environments set
+     */
+    public Set<String> setConsumersEnvironments(List<String> consumerUuids, List<String> targetEnvIds,
+        Owner owner) {
+
+        if (owner == null) {
+            throw new IllegalArgumentException("owner is null");
+        }
+
+        if (consumerUuids == null || consumerUuids.isEmpty() || targetEnvIds == null ||
+            targetEnvIds.isEmpty()) {
+            return new HashSet<>();
+        }
+
+        if (!ContentAccessMode.ORG_ENVIRONMENT.toDatabaseValue().equals(owner.getContentAccessMode())) {
+            throw new BadRequestException(i18n.tr("Owner is not in SCA content access mode"));
+        }
+
+        Set<String> unknownConsumerUuids = consumerCurator
+            .getNonExistentConsumerUuids(consumerUuids, owner.getKey());
+        if (!unknownConsumerUuids.isEmpty()) {
+            throw new BadRequestException(i18n.tr("Unkown consumer UUIDs: {0}",
+                unknownConsumerUuids));
+        }
+
+        Set<String> unknownEnvironmentIds = envCurator
+            .getNonExistentEnvironmentIds(targetEnvIds, owner);
+        if (!unknownEnvironmentIds.isEmpty()) {
+            throw new BadRequestException(i18n.tr("Unknown environment IDs: {0}",
+                unknownEnvironmentIds));
+        }
+
+        if (Util.containsDuplicateOrNull(targetEnvIds)) {
+            throw new BadRequestException(i18n.tr("Environment IDs contains duplicates or null value"));
+        }
+
+        log.info("setting {} consumers to {} environments",
+            consumerUuids.size(), targetEnvIds.size());
+
+        Set<String> consumersToUpdate = new HashSet<>();
+        Map<String, List<String>> consumerUuidToEnvs = envCurator.findEnvironmentsOf(consumerUuids);
+        for (String consumerUuid : consumerUuids) {
+            List<String> currentEnvs = consumerUuidToEnvs.get(consumerUuid);
+            if (currentEnvs == null || !targetEnvIds.equals(currentEnvs)) {
+                consumersToUpdate.add(consumerUuid);
+            }
+        }
+
+        log.info("{} of the provided consumers are not in the target environments currently",
+            consumersToUpdate.size());
+
+        if (consumersToUpdate.isEmpty()) {
+            return new HashSet<>();
+        }
+
+        int added = envCurator.setConsumersEnvironments(consumersToUpdate, targetEnvIds);
+        log.info("{} consumers have been set to the target environments", added);
+
+        // Delete all of the content access certificates for consumers that had an environment change
+        List<String> ids = caCertCurator.getIdsForConsumers(consumersToUpdate);
+        int unlinked = consumerCurator.unlinkCaCertificates(ids);
```

Comment:
Is it necessary to delete the whole cert, or just the accessible content json? Also, this should be performed by the ContentAccessManager (or perhaps the SCACertGenerator/Manager?), so we don't have to ask that question nor deal with the specifics of triggering it here.

Reply:
I agree with moving this operation to the ContentAccessManager, however an interesting problem occurs when moving the operation to the ContentAccessManager and injecting the ContentAccessManager into the ConsumerManager. 

The problem is that during startup, `ConsumerCheckInFilter` is being created which injects `ConsumerManager` and this class now injects `ContentAccessManager` and this class injects `EventSinkImpl` which has a `CandlepinRequestScoped` annotation. This throws the error:

```
1) [Guice/ErrorInCustomProvider]: OutOfScopeException: Cannot access Key[type=EventSinkImpl, annotation=[none]] outside of a scoping block
  at EventSinkImpl.class(EventSinkImpl.java:55)
  while locating EventSinkImpl
  at ContentAccessManager.<init>(ContentAccessManager.java:67)
      \_ for 4th parameter
  at ConsumerManager.<init>(ConsumerManager.java:60)
      \_ for 2nd parameter
  at ConsumerCheckInFilter.<init>(ConsumerCheckInFilter.java:50)
      \_ for 2nd parameter
  while locating ConsumerCheckInFilter
```

Reply:
Anything injecting a scoped or stateful object needs to be requested on demand via providers within the appropriate context and should not be injected directly.

If we're running into circular dependencies or are otherwise misusing our DI layer, that's something we should address sooner than later. Possibly as a prereq to this PR.

Reply:
Moved this operation to the `ContentAccessManager` and used the `Provider` for retrieving the `EventSinkImpl` in `ContentAccessManager`.